docs/data_types/video/video.md

```diff
@@ -0,0 +1,222 @@
+
+
+## VideoTensor
+
+DocArray offers several VideoTensors to store your data to:
+
+- [`VideoNdArray`][docarray.typing.tensor.video.video_ndarray.VideoNdArray]
+- `VideoTorchTensor`
+- `VideoTensorFlowTensor`
+
+If you specify the type of your tensor to one of the above, it will be cast to that automatically:
+
+```python hl_lines="7 8 15 16"
+from docarray import BaseDoc
+from docarray.typing import VideoTensorFlowTensor, VideoTorchTensor, VideoUrl
+
+
+class MyVideo(BaseDoc):
+    url: VideoUrl
+    tf_tensor: VideoTensorFlowTensor = None
+    torch_tensor: VideoTorchTensor = None
+
+
+doc = MyVideo(
+    url='https://github.com/docarray/docarray/blob/feat-rewrite-v2/tests/toydata/mov_bbb.mp4?raw=true'
+)
+
+doc.tf_tensor = doc.url.load().video
+doc.torch_tensor = doc.url.load().video
+
+assert isinstance(doc.tf_tensor, VideoTensorFlowTensor)
+assert isinstance(doc.torch_tensor, VideoTorchTensor)
+```
+
+
+
+## VideoBytes
+
+Alternatively, you can load your [`VideoUrl`][docarray.typing.url.VideoUrl] instance to [`VideoBytes`][docarray.typing.bytes.VideoBytes], and your [`VideoBytes`][docarray.typing.bytes.VideoBytes] instance to a `VideoTensor` of your choice:
+
+```python hl_lines="15 16"
+from docarray import BaseDoc
+from docarray.typing import VideoTensor, VideoUrl, VideoBytes
+
+
+class MyVideo(BaseDoc):
+    url: VideoUrl
+    bytes_: VideoBytes = None
+    video: VideoTensor = None
+
+
+doc = MyVideo(
+    url='https://github.com/docarray/docarray/blob/feat-rewrite-v2/tests/toydata/mov_bbb.mp4?raw=true'
+)
+
+doc.bytes_ = doc.url.load_bytes()
+doc.video = doc.url.load().video
+```
+ 
+Vice versa, you can also transform a [`VideoTensor`] to  [`VideoBytes`][docarray.typing.bytes.VideoBytes]:
+
+```python
+from docarray.typing import VideoBytes
+
+bytes_from_tensor = doc.video.to_bytes()
+
+assert isinstance(bytes_from_tensor, VideoBytes)
+```
+
+
+## Key frame extraction
+
+A key frame is defined as the starting point of any smooth transition.
+Given the key frame indices you can easily access selected scenes.
+
+```python
+indices = doc.key_frame_indices
+first_scene = doc.video[indices[0] : indices[1]]
+
+assert (indices == [0, 95]).all()
+assert first_scene.shape == (95, 176, 320, 3)
+```
+
+Or you can easily access the first frame of all new scenes and display them in a notebook:
+
+```python
+from docarray.typing import ImageNdArray
+from pydantic import parse_obj_as
+
+
+key_frames = doc.video[doc.key_frame_indices]
+for frame in key_frames:
+    img = parse_obj_as(ImageNdArray, frame)
+    img.display()
+```
+
+<figure markdown>
+  ![](key_frames.png){ width="350" }
+</figure>
+
+
+
+## Save video to file
+
+You can easily save your video tensor to a file. In the example below we save the video with a framerate of 60 fps, which results in a 4 sec video, instead of the original 10 second video with a frame rate of 25 fps. 
```

Comment:
```suggestion
You can save your video tensor to a file. In the example below you save the video with a framerate of 60 fps, which results in a 4 sec video, instead of the original 10 second video with a frame rate of 25 fps. 
```